spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java

```diff
@@ -35,6 +35,9 @@
 import org.apache.pulsar.client.api.SubscriptionInitialPosition;
 import org.apache.pulsar.client.api.SubscriptionType;
 import org.apache.pulsar.common.schema.SchemaType;
+import org.apache.pulsar.common.util.ObjectMapperFactory;
+import org.apache.pulsar.shade.com.fasterxml.jackson.core.JsonProcessingException;
```

Comment:
Lets use the non-shaded versions of these.

Reply:
That works for the `JsonProcessingException`, but the compiler complains when trying to replace the  shaded `ObjectMapper` coming from Pulsar's factory with the Jackson one

Reply:
Ah ok. We can get rid of that by not holding a reference and just do `ObjectMapperFactory.create().writeValueAsString` inline.

Reply:
In that case, the Exception type in the catch needs to be the shaded one again 😁